x/programs/rust/wasmlanche-sdk/src/program.rs

```diff
@@ -74,6 +74,25 @@ impl<K> Program<K> {
 
         borsh::from_slice(&bytes).map_err(|_| StateError::Deserialization)
     }
+
+    /// Gets the remaining fuel available to this program
+    /// # Errors
+    /// Returns a [`StateError`] if the call fails.
+    pub fn remaining_fuel(&self) -> Result<u64, StateError> {
+        #[link(wasm_import_module = "program")]
+        extern "C" {
+            #[link_name = "remaining_fuel"]
+            fn get_remaining_fuel() -> *const u8;
+        }
+
+        let bytes = {
+            let ptr = unsafe { get_remaining_fuel() };
+
+            deref_bytes(ptr)
+        };
+
+        borsh::from_slice::<u64>(&bytes).map_err(|_| StateError::Deserialization)
```

Comment:
Should we just use the native return instead of serializing deserializing? This method should be infallible. 

In any case, if we do want to do the whole serialization thing, we can just panic if deserialization fails.